### PR TITLE
support non-matching sample in evaluable.Sampled

### PR DESCRIPTION
**Pull request by gertjanvanzwieten**
Draft PR to discuss the desired API for functionality of this kind.

### CODE REVIEW

File: nutils/sample.py [hunk 1]
```diff
@@ -245,13 +245,13 @@ def __call__(self, __func: function.IntoArray) -> function.Array:
         indices = evaluable.loop_concatenate(evaluable._flat(self.get_evaluable_indices(ielem)), ielem)
         return _ReorderPoints(_ConcatenatePoints(func, self), indices) * funcscale
 
-    def basis(self) -> function.Array:
+    def basis(self, interpolation: str = 'none') -> function.Array:
```
Comment (joostvanzwieten) [inline nutils/sample.py:248]: Can you document the `interpolation` argument, e.g. by referring to `asfunction`?

File: nutils/sample.py [hunk 2]
```diff
@@ -245,13 +245,13 @@ def __call__(self, __func: function.IntoArray) -> function.Array:
...
         '''Basis-like function that for every point in the sample evaluates to the
         unit vector corresponding to its index.'''
 
         raise NotImplementedError
 
-    def asfunction(self, array: numpy.ndarray) -> function.Array:
+    def asfunction(self, array: numpy.ndarray, interpolation: str = 'none') -> function.Array:
```
Comment (joostvanzwieten) [inline nutils/sample.py:254]: Can you document the `interpolation` argument?

Comment (gertjanvanzwieten) [reply nutils/sample.py:260]: added "Interpolation scheme used to map sample values to the evaluating sample. Valid values are "none", demanding that the evaluating sample mathes self, or "nearest" for nearest-neghbour mapping."

Comment (joostvanzwieten) [reply nutils/sample.py:260]: Typo: 'neghbour' should be 'neighbour'.

Comment (gertjanvanzwieten) [reply nutils/sample.py:260]: fixed

**Review by joostvanzwieten — APPROVED** [review]

**Pull request merged**